lib/open_api_spex/cast/number.ex

```diff
@@ -0,0 +1,66 @@
+defmodule OpenApiSpex.Cast.Number do
+  @moduledoc false
+  alias OpenApiSpex.Cast
+
+  @spec cast(ctx :: Cast.t()) :: {:ok, Cast.t()} | Cast.Error.t()
+  def cast(%{value: value} = ctx) when is_number(value) do
+    case cast_number(ctx) do
+      {:cast, ctx} -> cast(ctx)
+      result -> result
+    end
+  end
+
+  # TODO We need a way to distinguish numbers in (JSON) body vs in request parameters
+  # so we can reject strings values for properties of `type: :number`
+  def cast(%{value: value} = ctx) when is_binary(value) do
+    case Float.parse(value) do
+      {value, ""} -> cast(%{ctx | value: value})
```

Comment:
Does this need to recurse to `cast/1`, or can it just continue on to `cast_number/1`?

Reply:
`number.ex` was copied almost directly from `integer.ex` and thus follows the same pattern.

As I understand it, the reason it has to call `cast(%Cast{})` is so the rest of the validation 'chain' can occur (the recursive calls to `cast/1` and `cast_number/1` all orchestrate to incrementally 'consume' the `minimum:` and `maximum:` validation properties in the schema).

In fact my original solution was a simpler, sequential one using a `with ... do` chain:

```
  def cast_number(%{value: value, schema: %{minimum: minimum, maximum: maximum}} = ctx)
      when not is_nil(minimum) or not is_nil(maximum) do

    with {:ok, value} <- cast_number(%{value: value}),
         {:ok, value} <- validate_minimum(value, minimum, ctx),
         {:ok, value} <- validate_maximum(value, maximum, ctx) do
      {:ok, value}
    end
  end

  defp validate_minimum(value, minimum, ctx)
       when is_float(minimum) or is_integer(minimum) do
    if value > minimum do
      {:ok, value}
    else
      OpenApiSpex.Cast.Error.new(ctx, {:minimum, minimum, value})
    end
  end

  defp validate_minimum(value, _, _), do: {:ok, value}
```

Let me know which one you prefer. I can also take a stab at simplifying `integer.ex` while am at it (or at some point in the future) to 'harmonize' the two again.

Reply:
I see that now. Thank you.